scipy/integrate/_quadrature.py

```diff
@@ -47,6 +49,20 @@ class AccuracyWarning(Warning):
     pass
 
 
+if TYPE_CHECKING:
+    # workaround for mypy function attributes see:
+    # https://github.com/python/mypy/issues/2087#issuecomment-462726600
+    from typing import Protocol, Dict, Tuple, Any
+    class CacheAttributes(Protocol):
+        cache: Dict[int, Tuple[Any, Any]]
+    def cache_decorator(func: Any) -> CacheAttributes:
```

Comment:
Probably want to make `func` at least `Callable`.

Reply:
Done